src/arc.ts

```diff
@@ -19,16 +17,21 @@ import {
   transactionErrorHandler,
   transactionResultHandler
 } from './operation'
-import { IProposalQueryOptions, Proposal } from './proposal'
+import { IProposalQueryOptions, Proposal } from './plugins/proposal'
 import { IRewardQueryOptions, Reward } from './reward'
-import { ISchemeQueryOptions, Scheme } from './scheme'
-import { SchemeBase } from './schemes/base'
 import { ABI_DIR } from './settings'
 import { IStakeQueryOptions, Stake } from './stake'
 import { ITagQueryOptions, Tag } from './tag'
 import { Token } from './token'
 import { Address, IPFSProvider, Web3Provider } from './types'
 import { isAddress } from './utils'
+import { IPFSClient } from './ipfsClient'
+import { IEventQueryOptions, Event } from './event'
+import { IPluginQueryOptions, Plugin } from './plugins/plugin'
+import { Plugins, Proposals } from './plugins'
+
+type PluginName = keyof typeof Plugins
```

Comment:
Love this! Let's move this to the `plugins/index.ts` file so others can re-use this.

Reply:
And the ProposalName as well

Reply:
Done